library/signed/src/Tests.qs

```diff
@@ -8,47 +8,24 @@ import Measurement.MeasureSignedInteger;
 /// This entrypoint runs tests for the signed integer library.
 operation Main() : Unit {
     UnsignedOpTests();
-    MeasureSignedIntTests();
+    Qtest.Operations.TestCases(MeasureSignedIntTests());
     SignedOpTests();
 
 }
 
-operation MeasureSignedIntTests() : Unit {
-    use a = Qubit[4];
-
-    // 0b0001 == 1
-    X(a[0]);
-    let res = MeasureSignedInteger(a, 4);
-    Fact(res == 1, $"Expected 1, received {res}");
-
-    // 0b1111 == -1
-    X(a[0]);
-    X(a[1]);
-    X(a[2]);
-    X(a[3]);
-    let res = MeasureSignedInteger(a, 4);
-    Fact(res == -1, $"Expected -1, received {res}");
-
-    // 0b01000 == 8
-    use a = Qubit[5];
-    X(a[3]);
-    let res = MeasureSignedInteger(a, 5);
-    Fact(res == 8, $"Expected 8, received {res}");
-
-    // 0b11110 == -2
-    X(a[1]);
-    X(a[2]);
-    X(a[3]);
-    X(a[4]);
-    let res = MeasureSignedInteger(a, 5);
-    Fact(res == -2, $"Expected -2, received {res}");
-
-    // 0b11000 == -8
-    X(a[3]);
-    X(a[4]);
-    let res = MeasureSignedInteger(a, 5);
-    Fact(res == -8, $"Expected -8, received {res}");
-
+function MeasureSignedIntTests() : (String, Int, (Qubit[]) => (), (Qubit[]) => Int, Int)[] {
```

Comment:
Minor comment: I know we have example in the docstrings. Possibly in the follow up PR, we could add some examples in `samples/testing` folder demonstrating `QTest` library in detail for ease of adoption.

Reply:
Given that this is just one library, I'm hesitant to enshrine it immediately. I wonder if others might want to contribute other test formats. That being said, I'm going to update more libraries that I've authored to use this testing pattern, and that'll hopefully be a good source of example code as well.

Reply:
> That being said, I'm going to update more libraries that I've authored to use this testing pattern, and that'll hopefully be a good source of example code as well.

I dont understand this quite well. I feel this is very similar to saying that if wants to see `pytest` usage, please see the code for XYZ python library/libraries which use `pytest` as a testing library.

I think `samples/testing` takes an outside view of Q#, giving users an insight to various ways of testing in Q#. I feel having `QTest` examples there should help to boost the adoption of `QTest` library.  Based on my prior experience with Q#, I sometimes find digging into Q# library, Q# internal rust/python tests, helpful. However, I feel that it shouldnt be an expectation.

However, since Q# is a new language and currently samples reside in the same github repo as the QDK repo which is another story(see #1934 ), probably converting existing library tests to use `QTest` library is not a bad option. I would like to have more thoughts on this. @swernli, @tcNickolas, @billti thoughts?